src/ctl.c

```diff
@@ -3457,3 +3467,35 @@ experimental_arenas_i_pactivep_ctl(tsd_t *tsd, const size_t *mib,
 	malloc_mutex_unlock(tsd_tsdn(tsd), &ctl_mtx);
 	return ret;
 }
+
+static int
+experimental_prof_recent_alloc_max_ctl(tsd_t *tsd, const size_t *mib,
+    size_t miblen, void *oldp, size_t *oldlenp, void *newp, size_t newlen) {
+	int ret;
+
+	if (!(config_prof && opt_prof)) {
+		ret = ENOENT;
+		goto label_return;
+	}
+
+	ssize_t old_max;
+
+	if (newp != NULL) {
+		ssize_t max;
+		WRITE(max, ssize_t);
+		if (max < -1) {
+			ret = EINVAL;
+			goto label_return;
+		}
+		old_max = prof_recent_alloc_max_set(tsd, max);
+	} else {
```

Comment:
nit: `else if (oldp != NULL)`. Also maybe remove the blank lines around the whole branch.

Reply:
To overcome the clang complain, I decided not to use `WRITE()` at all, which is actually the same as what's been done in the `mallctl` getter / setter for `prof_active`. There might be need for a better `WRITE()`. Will think about it.

The `else if (oldp != NULL)` might trigger the same clang complain though - it may think that `old_max` may not be initialized when being used (in `READ()`).